### DISCUSSION TITLE
TileLayer with WebSocket and createobjecturl?

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#5619] [Q&A] [answered]
I am using the [TileLayer](https://deck.gl/docs/api-reference/geo-layers/tile-layer) overriding the `getTileData` function which in turn calls a WebSocket connection to retrieve imagery. 

Currently the WebSocket returns an `arrayBuffer` with the image _(png)_ byte array. I am currently converting that `arrayBuffer` into an [object url](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL).

I am curious if deck is performing double-duty. Is deck taking the object url I return from the `getTileData` function and then re-fecthing it to convert it to an image? 

If that's the case, I am wondering if perhaps there is a more efficient approach I could be using that would not require deck to perform this extra fetching/conversion? Perhaps return the arrayBuffer directly from the `getTileData` function?

**Comment by Pessimistress** [comment, marked as answer]
You can use the [ImageLoader](https://loaders.gl/modules/images/docs/api-reference/image-loader) directly: https://deck.gl/docs/developer-guide/loading-data#example-parse-gltf-from-a-binary-blob
